### PR TITLE
Improve support for debugging signal emission

### PR DESCRIPTION
**Pull request by LinqLover**
With this patch, the following debugging scenarios do not raise a 'Must not emit signal from the outside' error any longer:

- Step over call to a signal method
- Step over call to the `#emit` method from a signal method
- Step over call to a wrapped signal method
- Step over call to a wrapped #emit method from a signal method (requires [Kernel-ct.1306 from the Inbox](http://forum.world.st/The-Inbox-Kernel-ct-1306-mcz-td5112342.html))

At the same time, I did not knowingly introduce any false positive errors.

(For a similar problem in the Squeak Trunk related to the wrapper contexts added by the debugger, see [Squeak - Dev - BUG/REGRESSION while debugging Generator >> #nextPut:](http://forum.world.st/BUG-REGRESSION-while-debugging-Generator-gt-gt-nextPut-td5108125.html). Maybe we should indeed find a general solution/protocol at some time for hiding such kind of private contexts from other users of the general context interface?)

## Testing

The following scenarios were tested in `SignalTest >> #test18SignalFromOutsideError`:

- [x] Step over call to a signal method
- [x] Step over call to the `#emit` method from a signal method
- [ ] Step over call to a wrapped signal method
- [ ] Step over call to a wrapped #emit method from a signal method

## Performance considerations

The average results of SignalBenchmarkSignals for this patch were `#(32.8 4.5)` as opposed to `#(37.0 3.25)` from 618c7d55d31ab6eb2dd679e77fdf23025f8295de. However, I'm afraid these benchmark values show a way too large dispersion with standard deviations of `#(11.5 1.3)`, _n = 4_, so at the end, I cannot tell whether these changes slow signal emission down significantly.

## Follow-up questions

I think this is a very interesting topic, so please forgive me some related questions: :-)

- Why are different kinds of method wrappers (MwMethodWrapper, SUnit, ...) only considered mutually exclusive (see the `dodged` temp var)? Won't this break when nesting method wrappers or calling one of them from a test, for example?
- Should there belong a `#todo` flag to the "handling" of calls from SmalltalkCI or is this insoluble?
- ~~Does this repository provide any tests for signal emission I could extend with a few regression tests? :-)~~ (just found them myself :D)

### CODE REVIEW

File: repository/Signals.package/monticello.meta/version [hunk 1]
```diff
@@ -1 +0,0 @@
...
... date '2 June 2010' time '10:53:54.289 pm' author 'mt' ancestors () stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())) stepChildren ())
```
Comment (LinqLover) [inline repository/Signals.package/monticello.meta/version:1]: Am, there were no contribution instructions in the readme. Did I do anything wrong by committing my changes using the Git Browser?

**Comment by LinqLover** [comment]
Update: As discussed, I won't write tests for method wrappers until something breaks indeed. Ready for review! 🎉

**Comment by marceltaeumel** [comment]
Hmm... why did you choose `#shouldFail:steppingOver:` instead of `#shouldntPass:steppingOver:`?

**Comment by LinqLover** [comment]
> Hmm... why did you choose `#shouldFail:steppingOver:` instead of `#shouldntPass:steppingOver:`?

Fair point. Shall I open another PR for that?
Thanks for merging! :-)

**Comment by marceltaeumel** [comment]
It's not worth it. Maybe later, if you find-fix another issue. ;-)

**Pull request merged**